python/deltalake/table.py

```diff
@@ -510,6 +510,30 @@ def version(self) -> int:
         """
         return self._table.version()
 
+    def partitions(
+        self,
+        partition_filters: Optional[List[Tuple[str, str, Any]]] = None,
+        as_tuple_list: bool = False,
+    ) -> List[Dict[str, str]] | List[Tuple[str]]:
```

Comment:
I would prefer if it's just one output type, list[dict] seems fine to be, people can always construct from that a list(tuple) if they want.

Reply:
Makes sense, updated PR. Thanks.